### PR TITLE
Add .NET Core package

### PR DESCRIPTION
**Pull request by cstrouse**
.NET Core is an open-source developer platform for building many different types
of applications.

Tested on x86_64.

### CODE REVIEW

File: packages/dotnet.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require 'package'
+
+class Dotnet < Package
+  description ''
```
Comment (uberhacker) [inline packages/dotnet.rb:4]: Missing description?

File: packages/dotnet.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    system "install -dm 755 #{CREW_DEST_PREFIX}/share/{dotnet,dotnet/packs,licenses}"
+    system "cp -dr --no-preserve='ownership' sdk templates #{CREW_DEST_PREFIX}/share/dotnet/"
+    system "export DOTNET_ROOT=#{CREW_PREFIX}/share/dotnet"
+    system "export PATH=$PATH:#{CREW_PREFIX}/share/dotnet"
+  end
+
+  def self.postinstall
+    puts
+    puts "All things dotnet are in #{CREW_PREFIX}/share/dotnet.".lightblue
+    puts
```
Comment (uberhacker) [inline packages/dotnet.rb:38]: Might be good to explain to the user this needs to be executed.  Something like:
```
puts "To complete the installation, execute the following:".lightblue
```

File: packages/dotnet_core.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  # icu, curl, glibc, openssl, zlib, kerberos5
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.install
+    system 'export DOTNET_CLI_TELEMETRY_OPTOUT=1'
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/dotnet"
```
Comment (uberhacker) [inline packages/dotnet_core.rb:28]: Doesn't `install` create the directory?  IOW, does it work without this line?

File: packages/dotnet_core.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    system "install -dm 755 #{CREW_DEST_PREFIX}/share/{dotnet,dotnet/packs,licenses}"
+    system "cp -dr --no-preserve='ownership' sdk templates #{CREW_DEST_PREFIX}/share/dotnet/"
+    system "export DOTNET_ROOT=#{CREW_PREFIX}/share/dotnet"
+    system "export PATH=$PATH:#{CREW_PREFIX}/share/dotnet"
```
Comment (uberhacker) [inline packages/dotnet_core.rb:32]: Why do we need these export lines?

File: packages/dotnet_core.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.install
+    system 'export DOTNET_CLI_TELEMETRY_OPTOUT=1'
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/dotnet"
+    system "install -dm 755 #{CREW_DEST_PREFIX}/share/{dotnet,dotnet/packs,licenses}"
+    system "cp -dr --no-preserve='ownership' sdk templates #{CREW_DEST_PREFIX}/share/dotnet/"
```
Comment (uberhacker) [inline packages/dotnet_core.rb:30]: This line could be Rubyized with `FileUtils.cp_r ...` instead.

File: packages/dotnet_core.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    system "export DOTNET_ROOT=#{CREW_PREFIX}/share/dotnet"
+    system "export PATH=$PATH:#{CREW_PREFIX}/share/dotnet"
+  end
+
+  def self.postinstall
+    puts
+    puts "All things dotnet are in #{CREW_PREFIX}/share/dotnet.".lightblue
+    puts
```
Comment (uberhacker) [inline packages/dotnet_core.rb:38]: Might be good to explain to the user this needs to be executed.  Something like:
```
puts "To complete the installation, execute the following:".lightblue
```

**Comment by uberhacker** [comment]
Thank you for the new packages @cstrouse!

File: packages/dotnet.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.install
+    system 'export DOTNET_CLI_TELEMETRY_OPTOUT=1'
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/dotnet"
+    system "install -dm 755 #{CREW_DEST_PREFIX}/share/{dotnet,dotnet/packs,licenses}"
+    system "cp -dr --no-preserve='ownership' sdk templates #{CREW_DEST_PREFIX}/share/dotnet/"
```
Comment (uberhacker) [inline packages/dotnet.rb:30]: This line could be Rubyized with `FileUtils.cp_r ...` instead.

File: packages/dotnet.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  # icu, curl, glibc, openssl, zlib, kerberos5
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.install
+    system 'export DOTNET_CLI_TELEMETRY_OPTOUT=1'
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/dotnet"
```
Comment (uberhacker) [inline packages/dotnet.rb:28]: Doesn't `install` create the directory?  IOW, does it work without this line?

File: packages/dotnet_core.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+  # icu, curl, glibc, openssl, zlib, kerberos5
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.install
+    system 'export DOTNET_CLI_TELEMETRY_OPTOUT=1'
```
Comment (uberhacker) [inline packages/dotnet_core.rb:27]: Why do we need this export line?

File: packages/dotnet.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+  # icu, curl, glibc, openssl, zlib, kerberos5
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  def self.install
+    system 'export DOTNET_CLI_TELEMETRY_OPTOUT=1'
```
Comment (uberhacker) [inline packages/dotnet.rb:27]: Why do we need this export line?

File: packages/dotnet_core.rb (hunk 3 above)
Comment (cstrouse) [reply packages/dotnet_core.rb:28]: Fails without this.

File: packages/dotnet.rb (hunk 2 above)
Comment (cstrouse) [reply packages/dotnet.rb:38]: Added

File: packages/dotnet.rb (hunk 1 above)
Comment (cstrouse) [reply packages/dotnet.rb:4]: Added. Forgot to add this change to the commit.

File: packages/dotnet_core.rb (hunk 4 above)
Comment (cstrouse) [reply packages/dotnet_core.rb:32]: Install docs say they're required.

File: packages/dotnet_core.rb (hunk 6 above)
Comment (cstrouse) [reply packages/dotnet_core.rb:38]: Added

**Comment by cstrouse** [comment]
Closing this for now. On a fresh install this doesn't even work and I don't remember what else I had to do to get this working.

**Pull request closed without merging**